internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -192,6 +197,36 @@ func (e *EnvStackConfig) Parameters() ([]*cloudformation.Parameter, error) {
 	return e.transformParameters(currParams, e.prevParams, transformEnvControllerParameters)
 }
 
+func getSecurityGroupConfig(mft *manifest.Environment) (*template.SecurityGroupConfig, error) {
+	if mft != nil && !mft.Network.VPC.SecurityGroupConfig.IsEmpty() {
+		if !mft.Network.VPC.SecurityGroupConfig.IsValid() {
+			var ingress = make([]template.SecurityGroupRules, len(mft.Network.VPC.SecurityGroupConfig.Ingress))
+			var egress = make([]template.SecurityGroupRules, len(mft.Network.VPC.SecurityGroupConfig.Egress))
+			for idx, ingressValue := range mft.Network.VPC.SecurityGroupConfig.Ingress {
+				ingress[idx].IpProtocol = ingressValue.IpProtocol
+				ingress[idx].CidrIp = ingressValue.CidrIp
+				ingress[idx].ToPort = ingressValue.ToPort
+				ingress[idx].FromPort = ingressValue.FromPort
+			}
+			for idx, egressValue := range mft.Network.VPC.SecurityGroupConfig.Egress {
+				egress[idx].IpProtocol = egressValue.IpProtocol
+				egress[idx].CidrIp = egressValue.CidrIp
+				egress[idx].ToPort = egressValue.ToPort
+				egress[idx].FromPort = egressValue.FromPort
+			}
+
+			return &template.SecurityGroupConfig{
+				Ingress: ingress,
+				Egress:  egress,
+			}, nil
+		} else {
+			return &template.SecurityGroupConfig{}, fmt.Errorf("security group config is invalid")
+		}
+	} else {
+		return &template.SecurityGroupConfig{}, nil
```

Comment:
was there a reason why you return an empty pointer? typically we'll return the zero value of the type so this would be:
```suggestion
		return nil, nil
```

Reply:
You are right. I will change this to nil.